### ISSUE TITLE
Pytest autouse fixture is reported as reportUnusedFunction

### ISSUE THREAD

**Issue by eli88fine** [microsoft/pyright#7469]
**Describe the bug**
A pytest fixture denoted as `autouse=True` is flagged as `reportUnusedFunction`.  The expected behavior would be to understand that this is most definitely used.

**Code or Screenshots**

```python
import pytest,os
from pytest_mock import MockerFixture
@pytest.fixture(autouse=True)
def _aws_credentials(mocker: MockerFixture) -> None:  # type: ignore[reportUnusedFunction] # this is an autouse fixture
    """Mock AWS Credentials to ensure real AWS is never accessed during unit tests."""
    mocker.patch.dict(
        os.environ,
        {
            "AWS_ACCESS_KEY_ID": "testing",
            "AWS_SECRET_ACCESS_KEY": "testing",
            "AWS_SECURITY_TOKEN": "testing",
            "AWS_SESSION_TOKEN": "testing",
            "AWS_DEFAULT_REGION": "us-east-1",
        },
    )
```

pytest-mock: v3.12
pytest: v8.0.2

**VS Code extension or command-line**
command line tool 1.1.352

**Comment by erictraut** [comment]
Pyright is working as designed here, so I don't consider this a bug. The function is named as though it's local to the file (not imported by any other file) and is not referenced anywhere within the file.

Your options are:
1. Rename your test function so it doesn't appear to pyright as though it's a private (local) function.
2. Disable or suppress the reportUnusedFunction diagnostic check.

**Comment by eli88fine** [comment]
ok, thanks for the explanation.  without the underscore, I got flagged by `ruff` https://docs.astral.sh/ruff/rules/pytest-missing-fixture-name-underscore/

I guess I just need to pick which tool I want to inline disable the rule for 🤷
